app/models/course.rb

```diff
@@ -307,6 +318,24 @@ def materials_list
     materials.where(workflow_state: 'chunked').distinct.pluck(:name)
   end
 
+  def create_missing_forum_imports(forum_ids)
+    filtered_forum_ids = forum_ids.reject do |forum_id|
+      forum_imports.exists?(imported_forum: forum_id)
+    end
+
+    Course::Forum.where(id: filtered_forum_ids).each do |forum|
+      forum_imports.build(imported_forum: forum)
+    end
+    save!
+  end
+
+  def nearest_forum_discussions(query_embedding)
+    imported_forum_discussions.connection.execute('SET hnsw.ef_search = 100')
```

Comment:
Looks like this can trigger race conditions when there are concurrent queries.

Reply:
Removed this line as the default hnsw search of 40 should be good enough for our use case.